mxtoai/tools/__init__.py

```diff
@@ -22,90 +25,126 @@
-    tool_mapping[ToolName.DELETE_SCHEDULED_TASKS] = DeleteScheduledTasksTool(context=context)
 
-    # LinkedIn tools (conditional on API key)
+def _initialize_linkedin_tools(context: RequestContext) -> dict[ToolName, Tool | None]:
+    """Initialize LinkedIn tools."""
+    tool_mapping = {}
     rapidapi_key = os.getenv("RAPIDAPI_KEY")
     if rapidapi_key:
         try:
             tool_mapping[ToolName.LINKEDIN_FRESH_DATA] = LinkedInFreshDataTool(api_key=rapidapi_key, context=context)
         except Exception:
-            # Silently fail and set to None - caller will handle logging
             tool_mapping[ToolName.LINKEDIN_FRESH_DATA] = None
-
         try:
             tool_mapping[ToolName.LINKEDIN_DATA_API] = LinkedInDataAPITool(api_key=rapidapi_key, context=context)
         except Exception:
-            # Silently fail and set to None - caller will handle logging
             tool_mapping[ToolName.LINKEDIN_DATA_API] = None
     else:
         tool_mapping[ToolName.LINKEDIN_FRESH_DATA] = None
         tool_mapping[ToolName.LINKEDIN_DATA_API] = None
+    return tool_mapping
+
+
+def create_tool_mapping(
+    context: RequestContext,
+    scheduled_tasks_tool_factory: Callable[[], Tool],
+    allowed_python_imports: list[str],
+    model: RoutedLiteLLMModel | None = None,
+) -> dict[ToolName, Tool | None]:
+    """
+    Create a mapping of ToolName enums to actual tool instances.
+
+    Args:
+        context: Request context for tools that need it
+        scheduled_tasks_tool_factory: Factory function to create limited scheduled tasks tool
+        allowed_python_imports: List of allowed Python imports for the interpreter
+        model: Optional RoutedLiteLLMModel instance for tools that need it
+
+    Returns:
+        dict[ToolName, Tool | None]: Mapping of tool names to instances
+
+    """
+    if model is None:
+        model = RoutedLiteLLMModel()
+
+    tool_mapping = {
+        ToolName.ATTACHMENT_PROCESSOR: AttachmentProcessingTool(context=context),
+        ToolName.CITATION_AWARE_VISIT: CitationAwareVisitTool(context=context),
+        ToolName.PYTHON_INTERPRETER: PythonInterpreterTool(authorized_imports=allowed_python_imports),
+        ToolName.WIKIPEDIA_SEARCH: WikipediaSearchTool(),
+        ToolName.REFERENCES_GENERATOR: ReferencesGeneratorTool(context=context),
+    }
+
+    try:
+        tool_mapping[ToolName.AZURE_VISUALIZER] = AzureVisualizerTool(model=model)
+    except Exception as e:
+        tool_mapping[ToolName.AZURE_VISUALIZER] = None
+        logger = get_logger("tools")
+        logger.warning(f"Failed to initialize AzureVisualizerTool: {e}")
+
+    tool_mapping.update(_initialize_search_tools(context))
+    tool_mapping.update(_initialize_linkedin_tools(context))
+
+    if os.getenv("JINA_API_KEY"):
+        tool_mapping[ToolName.DEEP_RESEARCH] = DeepResearchTool()
+    else:
+        tool_mapping[ToolName.DEEP_RESEARCH] = None
```

Comment:
For a TODO- let's remove deep research altogether from email_agent flow, there will be separate agent for it.